File: fun.cpp
#include <iostream>

int main() {
    char op;
    double num1, num2;

    std::cout << "Enter the first number: ";
    std::cin >> num1;

    std::cout << "Enter an operator (+, -, *, /): ";
    std::cin >> op;

    std::cout << "Enter the second number: ";
    std::cin >> num2;

    switch (op) {
        case '+':
            std::cout << "Result: " << num1 + num2 << std::endl;
            break;
        case '-':
            std::cout << "Result: " << num1 - num2 << std::endl;
            break;
        case '*':
            std::cout << "Result: " << num1 * num2 << std::endl;
            break;
        case '/':
            if (num2 != 0) {
                std::cout << "Result: " << num1 / num2 << std::endl;
            } else {
                std::cout << "Error: Division by zero is not allowed." << std::endl;
            }
            break;
        default:
            std::cout << "Error: Invalid operator." << std::endl;
            break;
    }

    return 0;
}

File: fun.m
#import <Foundation/Foundation.h>
#import <stdio.h>

int main(int argc, const char * argv[]) {
    @autoreleasepool {
        char inputBuffer[256];
        double num1, num2;
        char op;

        printf("Enter the first number: ");
        fgets(inputBuffer, sizeof(inputBuffer), stdin);
        num1 = atof(inputBuffer);

        printf("Enter an operator (+, -, *, /): ");
        fgets(inputBuffer, sizeof(inputBuffer), stdin);
        sscanf(inputBuffer, " %c", &op);

        printf("Enter the second number: ");
        fgets(inputBuffer, sizeof(inputBuffer), stdin);
        num2 = atof(inputBuffer);

        switch (op) {
            case '+':
                printf("Result: %f\n", num1 + num2);
                break;
            case '-':
                printf("Result: %f\n", num1 - num2);
                break;
            case '*':
                printf("Result: %f\n", num1 * num2);
                break;
            case '/':
                if (num2 != 0) {
                    printf("Result: %f\n", num1 / num2);
                } else {
                    printf("Error: Division by zero.\n");
                }
                break;
            default:
                printf("Error: Invalid operator.\n");
                break;
        }
    }
    return 0;
}
